COOP header breaks back/forward behavior when client-side redirects are involved

cdumez opened the pull request:
#### 268201eb1e9423b0fd8356e2071457334a48418f
<pre>
COOP header breaks back/forward behavior when client-side redirects are involved
<a href="https://bugs.webkit.org/show_bug.cgi?id=250781">https://bugs.webkit.org/show_bug.cgi?id=250781</a>
rdar://104659192

Reviewed by Brent Fulgham.

FrameLoader::loadPostRequest() was failing to set forward the LockBackForwardList
flag from the FrameLoadRequest to the NavigationAction. As a result, the UIProcess
would not be made aware and would fail to lock the back/forward list in case of
process-swap (e.g. due to the COOP HTTP header).

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::loadPostRequest):
* Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardListTests.mm:
(-[WKBackForwardNavigationDelegate webView:didReceiveAuthenticationChallenge:completionHandler:]):
(TEST):

Canonical link: <a href="https://commits.webkit.org/267553@main">https://commits.webkit.org/267553@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ef48d7f993732924ec000f20ed5bbac3037563c0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16882 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17205 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17662 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18665 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15806 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17072 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20437 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17343 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18065 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17081 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17438 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14617 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19472 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14681 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15310 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22042 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15671 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15477 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19788 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16081 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13633 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15249 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4060 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19611 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15912 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
I wonder if we need a helper to make sure we populate from a request more consistently.

But this change looks great.

cdumez committed:
COOP header breaks back/forward behavior when client-side redirects are involved
https://bugs.webkit.org/show_bug.cgi?id=250781
rdar://104659192

Reviewed by Brent Fulgham.

FrameLoader::loadPostRequest() was failing to set forward the LockBackForwardList
flag from the FrameLoadRequest to the NavigationAction. As a result, the UIProcess
would not be made aware and would fail to lock the back/forward list in case of
process-swap (e.g. due to the COOP HTTP header).

* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::loadPostRequest):
* Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardListTests.mm:
(-[WKBackForwardNavigationDelegate webView:didReceiveAuthenticationChallenge:completionHandler:]):
(TEST):

Canonical link: https://commits.webkit.org/267553@main
